modules/BTBidAdapter.js

```diff
@@ -0,0 +1,237 @@
+import { config } from '../src/config.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { isBoolean, isFn, isStr } from '../src/utils.js';
+import { BANNER } from '../src/mediaTypes.js';
+
+const BIDDER_CODE = 'blockthrough';
+const GVLID = 815;
+const CURRENCY = config.getConfig('currency.adServerCurrency') || 'USD';
+const ENDPOINT_URL = 'https://pbs.btloader.com/openrtb2/auction';
+
+export const spec = {
+  code: BIDDER_CODE,
+  gvlid: GVLID,
+  supportedMediaTypes: [BANNER],
+  isBidRequestValid: (bid) => {
```

Comment:
might be worth adding JSDoc, e.g.:
<img width="883" alt="image" src="https://github.com/blockthrough/Prebid.js/assets/30797461/1c565aef-df18-4918-8528-3c637c4a7949">

it's up to you, however I prefer to declare methods separately to make export `spec` clearer:
```
function isBidRequestValid() { ... }
function buildRequests() { ... }

export const spec = {
  code: BIDDER_CODE,
  gvlid: GVLID,
  supportedMediaTypes: [BANNER],
  isBidRequestValid,
  buildRequests,
  ...
}
```